src/views/AccountDrefApplications/DrefTableActions/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { useContext, useMemo } from 'react';
+import { generatePath } from 'react-router-dom';
+import {
+    AddLineIcon,
+    CaseManagementIcon,
+    CheckLineIcon,
+    DocumentPdfLineIcon,
+    DownloadLineIcon,
+    PencilLineIcon,
+    ShareLineIcon,
+} from '@ifrc-go/icons';
+
+import DropdownMenuItem from '#components/DropdownMenuItem';
+import TableActions from '#components/Table/TableActions';
+import Link from '#components/Link';
+import useTranslation from '#hooks/useTranslation';
+import RouteContext from '#contexts/route';
+import { components } from '#generated/types';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+const DREF_STATUS_COMPLETED = 1;
+const DREF_STATUS_IN_PROGRESS = 0;
+
+export interface Props {
+    id: number;
+    status: components['schemas']['OperationTypeEnum'] | null | undefined;
+
+    // FIXME: typings should be fixed in the server
+    // Should be DREF | OPS_UPDATE | FINAL_REPORT
+    applicationType: string;
+    canAddOpsUpdate: boolean;
+    canCreateFinalReport: boolean;
+}
+
+function DrefTableActions(props: Props) {
+    const {
+        id,
+        status,
+        applicationType,
+        canAddOpsUpdate,
+        canCreateFinalReport,
+    } = props;
+
+    const strings = useTranslation(i18n);
+
+    const {
+        drefApplicationForm: drefApplicationFormRoute,
+    } = useContext(RouteContext);
+
+    const editPath = useMemo(
+        () => {
+            if (applicationType === 'DREF') {
+                return generatePath(drefApplicationFormRoute.absolutePath, { drefId: id });
+            }
+
+            return undefined;
+        },
+        [id, applicationType, drefApplicationFormRoute],
+    );
```

Comment:
Why not just write this directly on render?

Reply:
We'll have 2 more type of links for the forms (Ops Update and Final Report) I'll add TODOs